src/main/java/uk/gov/pay/api/filter/ratelimit/RateLimiter.java

```diff
@@ -12,50 +13,51 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(RateLimiter.class);
 
-    private final int rate;
-    private final int rateForPost;
+    private int rate;
+    private int rateForPost;
+    private int auditRate;
+    private int perMillis;
+    private int noOfNodes;
     
     private static final String METHOD_POST = "POST";
     
-    private final int auditRate;
-    private final int perMillis;
-    private final Cache<String, RateLimit> cache;
     private final Cache<String, RateLimit> auditCache;
-
-    public RateLimiter(int rate, int rateForPost, int auditRate, int perMillis) {
+    LocalRateLimiter localRateLimiter;
+    RedisRateLimiter redisRateLimiter;
+    
+    @Inject(optional = true)
+    public RateLimiter(int rate, int rateForPost, int auditRate, int perMillis, int noOfNodes,
+                       LocalRateLimiter localRateLimiter, RedisRateLimiter redisRateLimiter) {
+        
         this.rate = rate;
         this.rateForPost = rateForPost;
         this.auditRate = auditRate;
         this.perMillis = perMillis;
-        this.cache = CacheBuilder.newBuilder()
-                .expireAfterAccess(perMillis, TimeUnit.MILLISECONDS)
-                .build();
-
+        this.noOfNodes = noOfNodes;
+        
         this.auditCache = CacheBuilder.newBuilder()
                 .expireAfterAccess(perMillis, TimeUnit.MILLISECONDS)
                 .build();
+        
+        this.localRateLimiter = localRateLimiter;
+        this.redisRateLimiter = redisRateLimiter;
     }
 
-    void checkRateOf(String key, String method) throws RateLimitException {
+    public void checkRateOf(String key, String method) throws RateLimitException {
         try {
-            cache.get(key, () -> new RateLimit(getRateForMethod(method), perMillis)).updateAllowance();
-        } catch (ExecutionException e) {
-            //ExecutionException is thrown when the valueLoader throws a checked exception.
-            //We just create a new instance so no exceptions will be thrown, this should never happen.
-            LOGGER.error("Unexpected error creating a Rate Limiter object in cache", e);
+            redisRateLimiter.checkRateOf(key, getRateForMethod(method), perMillis);
+        } catch (RedisRateLimitException e) {
+            LOGGER.warn("Exception occurred checking rate limits using RedisRateLimiter, falling back to LocalRateLimiter");
+            
+            /* Used as fail safe mechanism in case redis is unavailable. 
+               Local rate Limiting should not replace redis for accurate rate limiting */
```

Comment:
i don't think this comment adds value as it's obvious we're falling back to local rate limiting here

Reply:
Agree. Removed comment